test/unit/test_spike_source.cpp

```diff
@@ -115,3 +115,41 @@ TEST(spike_source, exhaust)
     test_seq(regular_schedule(0, 1, 5));
     test_seq(explicit_schedule({0.3, 2.3, 4.7}));
 }
+
+TEST(spike_source, multiple)
+{
+    // This test assumes that seq will exhaust itself before t=10 ms.
+    auto test_seq = [](auto&&... seqs) {
+        ss_recipe rec(1u, spike_source_cell("src", static_cast<decltype(seqs)&&>(seqs)...));
+        cell_label_range srcs, tgts;
+        spike_source_cell_group group({0}, rec, srcs, tgts);
+
+        // epoch ending at 10ms
+        epoch ep(0, 0., 10.);
+        group.advance(ep, 1, {});
+
+        auto expected = spike_times(group.spikes());
+        std::sort(expected.begin(), expected.end());
+
+        auto actual = std::vector<time_type>{};
+        std::vector<schedule> schedules{seqs...};
```

Comment:
this should probably in the beginning of the lambda, such that it copies `seqs` before a potential move. It could be correct, if the (potential) move does not actually move anything and leaves the `seqs` in a good state. Haven't checked though.

Reply:
In the way of things it's fine, we keep one more copy alive then we should, but it's really not relevant, especially not in a test. I'd even argue it would be fine anywhere else.

Reply:
The advantage of perfect forwarding in the lambda is that you can stick in either l-value or r-value references and, thus, test both variants. If you feel this is unnecessary, we can leave it as is. I would prefer testing both variants by adding another call to the lambda:
```
auto req_sched = regular_schedule(0, 1, 5)
test_seq(req_sched, explicit_schedule({0.3, 2.3, 4.7}));
```

Reply:
I am not opposed to adding this, but what are we exactly testing? The removal of the copy ctor? We cannot observe
whether that occurs or not, since we do not instrument the ctors.

Reply:
We would be using the move constructor, instead. But you are right, there is not much to test there. If you pass by value to the `spike_source_cell` constructor, you can also remove one of the tests where you pass an l-value ref of a vector.

Reply:
I like the perfect forwarding constructor and the extra test doesn't hurt.